### PR TITLE
UCP: fix pack/unpack of the priority and caps flags in the address.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by yosefe** [comment]
nice catch, pls add unit test that would have caught this issue

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1479/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3401/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_wireup.cc [hunk 1]
```diff
@@ -248,12 +248,20 @@ UCS_TEST_P(test_ucp_wireup, address) {
...
     status = ucp_address_pack(sender().worker(), NULL, -1, order, &size, &buffer);
     ASSERT_UCS_OK(status);
     ASSERT_TRUE(buffer != NULL);
     ASSERT_GT(size, 0ul);
     EXPECT_LE(size, 512ul); /* Expect a reasonable address size */
+    for (tl = 0; tl < sender().worker()->context->num_tls; tl++)
+    {
+        /* Make sure that the packed priority in the address is valid.
+         * 40 is the current highest value for a device priority */
+        EXPECT_LE(sender().worker()->iface_attrs[tl].priority, 40);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_wireup.cc]: need to take the value '40' from interface attributes on the worker

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1480/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3402/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1483/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3405/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1487/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3409/ for details (Mellanox internal link).

**Pull request merged**